pkg/ebpf/events_derived.go

```diff
@@ -81,11 +53,19 @@ func (t *Tracee) deriveEvents(ctx context.Context, in <-chan *trace.Event) (<-ch
 				out <- event
 
 				// Derive event before parsing its arguments
-				derivatives := t.deriveEvent(*event)
+				derivatives, errors := events.Derive(*event, t.eventDerivations)
 
 				for _, derivative := range derivatives {
+					// Drop derived events which were not requested by the user
+					if !t.events[events.ID(derivative.EventID)].emit {
+						continue
+					}
 					out <- &derivative
 				}
+
+				for _, err := range errors {
+					t.handleError(err)
+				}
```

Comment:
No, but there is a problem of this being inside the loop, which i'll fix.

Reply:
NVM, it's not inside the loop. I think it's slightly better to do them after the event loop so the events get to the next processing stage ASAP. But I don't mind moving it before if you think it's better.

Reply:
I think it is best to check for errors right after the return because:
1. It is more readable
2. To avoid future bugs - If someone will add some new logic between the function and the error check that uses err again